pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -239,6 +239,10 @@ func (m *kubeGenericRuntimeManager) startContainer(ctx context.Context, podSandb
 		}
 	}
 
+	if restartCount > 0 {
+		m.recordContainerEvent(pod, container, "", v1.EventTypeNormal, events.RestartedContainer, fmt.Sprintf("Restarting container %s", container.Name))
```

Comment:
It seems weird to me to emit both a RestartedContainer event and a StartedContainer event (on line 291). It also seems confusing that these are emitted at different points.

Reply:
Tim, I am ok to differentiate restart from the initial start event, but /hold the PR for other inputs.

Reply:
The change added by the PR actually is for when the container dies and restart process **begin** so it shows the container is currently **Restarting..** but restart hasn't been yet successful. Adding `Restarted` as reason is beneficial as in prior discussion, addition of new container state reason wasn't accepted also it will show how many times the container restarted which essentially serves the purpose.